tutor/src/components/task-step/exercise.cjsx

```diff
@@ -123,27 +123,21 @@ module.exports = React.createClass
     @setCurrentStep(current)
 
   setCurrentStep: (currentStep) ->
-    return unless currentStep isnt @state.currentStep
-
+    return if currentStep is @state.currentStep
     @setState({currentStep})
     @props.goToStep(currentStep)
 
-  setCurrentStepByStepId: (id) ->
-    {taskId} = @props
-    stepNavIndex = TaskStore.getStepNavIndex(taskId, id)
-    @setCurrentStep(stepNavIndex)
-
-  onFreeResponseChange: (id, tempFreeResponse) ->
+  onFreeResponseChange: (id, tempFreeResponse, stepIndex) ->
     TaskStepActions.updateTempFreeResponse(id, tempFreeResponse)
 
     # set part to be active if part of multipart
-    @setCurrentStepByStepId(id) unless @isSinglePart(@state.parts)
+    @setCurrentStep(stepIndex) unless @isSinglePart(@state.parts)
```

Comment:
Hm.  You may be correct but I hope not.

The current bug is because what `Exercise` was getting from `TaskStore.getStepNavIndex` was incorrect when it was for readings.  That would cause it to set the wrong step as active every time a key was pressed and the `onFreeResponseChange` callback occurred, and then as part of the of setting the step active it would scroll to it, causing the page to jump around.

What I'm attempting to do is to have `Exercise tell each of the exercises what their index is, and then they pass that back to the `Exercise` callbacks.  

I think `stepIndex` is the thing to use even with the interstitial cards, since it's derived from the `step.index` from the task, so they don't know anything about the cards.

At any rate, let's hold off for #1354 and see if it breaks.